File: typescript/rna-transcription/rna-transcription.ts
export function toRna(strand: string): string {
  return strand
    .split("")
    .map(fromDnaToRna)
    .join("");
}

const DNA_TO_RNA: Record<string, string> = {
  C: "G",
  G: "C",
  A: "U",
  T: "A"
} as const

function fromDnaToRna(nucleotide: string): string {
  const translated = DNA_TO_RNA[nucleotide];
  
  if(translated) {
    return translated;
  }
  
  throw new Error("Invalid input DNA.");
}

File: go/rna-transcription/rna_transcription.go
package strand

import "strings"

// ToRNA transcribes a DNA string into an RNA string by applying
// the standard nucleotide complement rule (A → U, T → A, C → G, G → C).
// It does not validate the DNA string or handle invalid nucleotides.
func ToRNA(dna string) string {
	return strings.Map(getNucleotideComplement, dna)
}

// getNucleotideComplement returns the RNA complement for a given DNA nucleotide.
func getNucleotideComplement(nucleotide rune) rune {
	return nucleotidesComplements[nucleotide]
}

var nucleotidesComplements = map[rune]rune{
	'C': 'G',
	'G': 'C',
	'A': 'U',
	'T': 'A',
}
